### PR TITLE
Remove treat support

### PR DESCRIPTION
**Pull request by askoufis**

### CODE REVIEW

File: packages/sku/config/eslint/importOrderConfig.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ module.exports = {
         ],
         pathGroups: [
           {
-            pattern: '*.+(treat|less|css)',
+            pattern: '*.+(less|css)',
```
Comment (mrm007) [inline packages/sku/config/eslint/importOrderConfig.js:32]:
```suggestion
            pattern: '*.@(less|css)',
```

https://github.com/micromatch/picomatch#extglobs

File: packages/sku/config/webpack/plugins/sku-webpack-plugin/index.js [hunk 2]
```diff
@@ -65,7 +63,7 @@ class SkuWebpackPlugin {
         include: this.include,
         oneOf: [
           {
-            compiler: /(vanilla-extract|treat-webpack-loader)/,
+            compiler: /(vanilla-extract)/,
```
Comment (mrm007) [inline packages/sku/config/webpack/plugins/sku-webpack-plugin/index.js:66]:
```suggestion
            compiler: /\bvanilla-extract\b/,
```

File: packages/sku/lib/validatePeerDeps.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ const asyncMap = (list, fn) => {
   return Promise.all(list.map((item) => fn(item)));
 };
 
-const singletonPackages = ['react-treat'];
+const singletonPackages = ['@vanilla-extract/css'];
```
Comment (mrm007) [inline packages/sku/lib/validatePeerDeps.js:20]: 👍

File: packages/sku/scripts/postinstall.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ if (packageJsonExists) {
   const skipPostInstall = skuSkipPostInstall || skuSkipPostinstall;
   const hasSku =
     Boolean(devDependencies?.sku) ||
-    // TODO: get rid of this part when we remove treat
+    // sku should always be a dev dependency, but some repos may still have it as a regular dependency
```
Comment (mrm007) [inline packages/sku/scripts/postinstall.js:33]: Should we warn them? It would somewhat reduce the size of their Docker image 🌍🔥

**Review by mrm007 — APPROVED** [review]
Goodbye 🍬

File: packages/sku/scripts/postinstall.js (hunk 4 above)
Comment (askoufis) [reply packages/sku/scripts/postinstall.js:33]: That's a good idea. More warning banners 🎉

Comment (mrm007) [reply packages/sku/scripts/postinstall.js:33]: jokes

File: packages/sku/config/webpack/plugins/sku-webpack-plugin/index.js (hunk 2 above)
Comment (askoufis) [reply packages/sku/config/webpack/plugins/sku-webpack-plugin/index.js:66]: Do we even need the brackets anymore, since there's only one variant?

Comment (mrm007) [reply packages/sku/config/webpack/plugins/sku-webpack-plugin/index.js:66]: I replaced the brackets with `\b`s

Comment (askoufis) [reply packages/sku/config/webpack/plugins/sku-webpack-plugin/index.js:66]: Oh, I though `\b` was brackets, it's word boundary 🤦

File: packages/sku/scripts/postinstall.js (hunk 4 above)
Comment (askoufis) [reply packages/sku/scripts/postinstall.js]: https://github.com/seek-oss/sku/pull/809/commits/64c4d37903975c73d8acdd8c5c58b44bcfc44a41

**Comment by askoufis** [comment]
This is blocked by #743. Braid v31 still references `sku/react-treat`, so we need to get on v32 before we can remove treat.

**Pull request merged**